algebra/src/curves/sw6/mod.rs

```diff
@@ -146,7 +146,7 @@ impl SW6 {
         let mut elt_q3 = elt.clone();
         elt_q3.frobenius_map(3);
         // elt_q3_over_elt = elt^(q^3-1)
-        let elt_q3_over_elt = elt_q3 * &elt_inv;
+        let elt_q3_over_elt = elt_q3 * elt_inv;
```

Comment:
Was this failing to compile after you added the new impls?

Reply:
Yes, with:
```rust
error[E0277]: cannot multiply `&&fields::models::fp6_2over3::Fp6<fields::sw6::fq6::Fq6Parameters>` to `fields::models::fp6_2over3::Fp6<fields::sw6::fq6::Fq6Parameters>`
   --> algebra/src/curves/sw6/mod.rs:149:38
    |
149 |         let elt_q3_over_elt = elt_q3 * &elt_inv;
    |                                      ^ no implementation for `fields::models::fp6_2over3::Fp6<fields::sw6::fq6::Fq6Parameters> * &&fields::models::fp6_2over3::Fp6<fields::sw6::fq6::Fq6Parameters>`
    |
    = help: the trait `std::ops::Mul<&&fields::models::fp6_2over3::Fp6<fields::sw6::fq6::Fq6Parameters>>` is not implemented for `fields::models::fp6_2over3::Fp6<fields::sw6::fq6::Fq6Parameters>`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
```

Note that `elt_inv: &Fq6`, so that I think the trait bound is `&Fq6: Mul<&'a $0>`, which would not match in the presence of `impl<P: Fp6Parameters> Mul<Fp6<P>> for Fp6<P>`. IIUC, in its absence (i.e. before this commit), autoref would fire, leading the trait system to solve for `Fq6: Mul<&'a $0>` which matches the (single) existing `impl Mul`.

(Yes, I'm preying for the success of rust-lang issue 44619)